coil-base/src/main/java/coil/drawable/CrossfadeDrawable.kt

```diff
@@ -43,13 +43,15 @@ class CrossfadeDrawable @JvmOverloads constructor(
     private val intrinsicWidth = computeIntrinsicDimension(start?.intrinsicWidth, end?.intrinsicWidth)
     private val intrinsicHeight = computeIntrinsicDimension(start?.intrinsicHeight, end?.intrinsicHeight)
 
+    private var start: Drawable? = null
     private var startTimeMillis = 0L
     private var maxAlpha = 255
     private var state = STATE_START
 
     init {
         require(durationMillis > 0) { "durationMillis must be > 0." }
 
+        this.start = start?.mutate()
         start?.callback = this
         end?.callback = this
```

Comment:
Should this be `this.start` and `this.end`? `drawable.mutate` is supposed to return the same drawable, but there's nothing to prevent a bad drawable from returning a new drawable.